src/main/java/uk/org/okapibarcode/output/Java2DRenderer.java

```diff
@@ -67,18 +71,61 @@ public class Java2DRenderer implements SymbolRenderer {
      * @param ink the ink (foreground) color
      */
     public Java2DRenderer(Graphics2D g2d, double magnification, Color paper, Color ink) {
+        this(g2d, magnification, paper, ink, 0);
+    }
+
+    /**
+     * Creates a new Java 2D renderer with the specified clockwise rotation. If the specified paper color is {@code null}, the symbol is drawn without clearing the
+     * existing {@code g2d} background.
+     *
+     * @param g2d the graphics to render to
+     * @param magnification the magnification factor to apply
+     * @param paper the paper (background) color (may be {@code null})
+     * @param ink the ink (foreground) color
+     * @param rotation the clockwise rotation of the symbol in degrees (0, 90, 180, or 270)
+     */
+    public Java2DRenderer(Graphics2D g2d, double magnification, Color paper, Color ink, int rotation) {
         this.g2d = g2d;
         this.magnification = magnification;
         this.paper = paper;
         this.ink = ink;
+        this.rotation = SymbolRenderer.normalizeRotation(rotation);
     }
 
     /** {@inheritDoc} */
     @Override
     public void render(Symbol symbol) {
-
+        int width = (int) (symbol.getWidth() * magnification);
+        int height = (int) (symbol.getHeight() * magnification);
         int marginX = (int) (symbol.getQuietZoneHorizontal() * magnification);
         int marginY = (int) (symbol.getQuietZoneVertical() * magnification);
+        int rotateHeight = height;
+        int rotateWidth = width;
+
+        // render rotation counter-clockwise
```

Comment:
Per all the other documentation, we should be rotating clockwise, not counter-clockwise.